rviz_default_plugins/src/rviz_default_plugins/displays/wrench/wrench_display.cpp

```diff
@@ -127,11 +133,24 @@ bool validateFloats(const geometry_msgs::msg::WrenchStamped & msg)
 
 void WrenchDisplay::processMessage(geometry_msgs::msg::WrenchStamped::ConstSharedPtr msg)
 {
-  if (!validateFloats(*msg)) {
-    setStatus(
-      rviz_common::properties::StatusProperty::Error, "Topic",
-      "Message contained invalid floating point values (nans or infs)");
-    return;
+  auto custom_msg = std::make_shared<geometry_msgs::msg::WrenchStamped>();
+  bool accept_NaN = accept_NaN_values_->getBool();
+
+  if (!accept_NaN) {
+    if (!validateFloats(*msg)) {
+      setStatus(
+        rviz_common::properties::StatusProperty::Error, "Topic",
+        "Message contained invalid floating point values (nans or infs)");
+      return;
+    }
+  } else {
+      custom_msg->wrench.force.x = (std::isnan(msg->wrench.force.x)) ?  0.0 : msg->wrench.force.x;
```

Comment:
You should also validate if message has inf in it. Use again validate for this and then if true check if there are only NaNs and if not than do nothing/ignore message.

Reply:
sure, will add the validate part,